portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -100,27 +106,35 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     commentsEntity.setProperty("email", email);
     commentsEntity.setProperty("message", comment);
     commentsEntity.setProperty("timestamp", timestamp);
+    commentsEntity.setProperty("userId", userId);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentsEntity);
     
-    // Redirect back to the HTML page.
-    response.sendRedirect("/index.html");
+    // Redirect back to the jsp page.
+    response.sendRedirect("/");
   }
 
   @Override
   public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Comment");
 
+    UserService userService = UserServiceFactory.getUserService();
+    String userId = userService.getCurrentUser().getUserId();
+    
     if (request.getParameter("id").equals("all")) {
-        
-        Query query = new Query("Comment");
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         PreparedQuery results = datastore.prepare(query);
 
         for (Entity entity : results.asIterable()) {
-            datastore.delete(entity.getKey());
+            String commentUserId = (String) entity.getProperty("userId");
+            if (userId.equals(commentUserId)){
```

Comment:
Alternatively, let the database do the work by filtering the original query down to the comments owned by the logged-in user.

Reply:
That's smart, I'll do that. Thanks!!